csvy/validators/table_schema.py

```diff
@@ -0,0 +1,454 @@
+        name: The name of the column.
+        title: A nicer human readable label or title for the field.
+        example: An example value for the field.
+        description: A description for the field.
+        constraints: A dictionary of constraints for the field.
+
+    """
+
+    name: str = Field(..., description="Column name.")
+    title: str | None = Field(
+        None, description="A nicer human readable label or title for the field."
+    )
+    example: str | None = Field(None, description="An example value for the field.")
+    description: str | None = Field(None, description="A description for the field.")
+    constraints: ConstraintsValidator = Field(
+        ConstraintsValidator(), description="A dictionary of constraints for the field."
+    )
+
+    def model_dump(self, *args, **kwargs) -> dict[str, Any]:
+        """Dump the model to a dictionary.
+
+        This method dumps the model to a dictionary. It sets exclude_unset to True and
+        by_alias to True, so that only the attributes that were set are included in the
+        dictionary and their aliases are always used.
+
+        Finally, it converts the attributes that are Enum instances to their values.
+
+        Returns:
+            A dictionary with the model attributes.
+
+        """
+        kwargs["exclude_unset"] = True
+        kwargs["by_alias"] = True
+        output = super().model_dump(*args, **kwargs)
+        for key, value in output.items():
+            if isinstance(value, Enum):
+                output[key] = value.value
+        if "constraints" in output and not output["constraints"]:
+            del output["constraints"]
+        return output
+
+
+class DefaultColumnValidator(CommonValidator):
+    """Default column validator for anything that does not have a specific one.
+
+    Attributes:
+        type_: A string specifying the type, valid values are any of the TypeEnum that
+            do not have a specific validator.
+        format_: A string specifying a format, with valid values being 'default' and
+            None.
+
+    """
+
+    type_: Literal[
+        TypeEnum.OBJECT,
+        TypeEnum.ARRAY,
+        TypeEnum.YEAR,
+        TypeEnum.YEARMONTH,
+        TypeEnum.DURATION,
+        TypeEnum.ANY,
+        None,
+    ] = Field(None, alias="type", description="A string specifying the type.")
+    format_: FormatDefault | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+
+
+class StringColumnValidator(CommonValidator):
+    """Validator for the string columns in the Table Schema.
+
+    This class is used to validate the string columns in the Table Schema. It is based
+    on the columns defined in the Table Schema specification.
+
+    Attributes:
+        type_: A string specifying the type, with valid values being 'string'.
+        format_: A string specifying a format, with valid values being 'default',
+            'email', 'uri', 'binary', 'uuid' and None.
```

Comment:
This "with valid values..." comment is made in multiple docstrings and is essentially just referencing the values in the Enum used. If these Enums ever change these docstring will also need to be updated, which can easily be missed. Maybe it's best to just reference the Enum? This can be done quite seamlessly with a link in the docs if we're using mkdocstrings.

Reply:
You're right. I've open a separate issue about that (https://github.com/ImperialCollegeLondon/pycsvy/issues/170) as we do not have proper documentation, yet.